pyiron_contrib/tinybase/creator.py

```diff
@@ -0,0 +1,255 @@
+        if name in self._jobs:
+            if task != self._jobs[name]:
+                raise ValueError(
+                    "Refusing to register task: different task of the same name already exists!"
+                )
+            else:
+                return self._jobs[name]
+        self._jobs[name] = task
+        return task
+
+
+class StructureCreator(Creator):
+    def __init__(self, project, config):
+        pass
+
+    @wraps(ase.build.bulk)
+    def bulk(self, *args, **kwargs):
+        return ase_to_pyiron(ase.build.bulk(*args, **kwargs))
+
+    @wraps(Atoms)
+    def atoms(self, *args, **kwargs):
+        return Atoms(*args, **kwargs)
+
+
+class ExecutorCreator(Creator):
+    _DEFAULT_CPUS = min(int(0.5 * len(sched_getaffinity(0))), 8)
+
+    def __init__(self, project, config):
+        self._most_recent = None
+
+    def most_recent(self):
+        if self._most_recent is None:
+            self._most_recent = Executor()
+        return self._most_recent
+
+    def _save(func):
+        @wraps(func)
+        def f(self, *args, **kwargs):
+            self._most_recent = func(self, *args, **kwargs)
+            return self._most_recent
+
+        return f
+
+    @wraps(ProcessExecutor)
+    @_save
+    def process(self, max_processes=_DEFAULT_CPUS):
+        return ProcessExecutor(max_processes=max_processes)
+
+    @wraps(BackgroundExecutor)
+    @_save
+    def background(self, max_threads=4):
+        return BackgroundExecutor(max_threads=max_threads)
+
+    @wraps(DaskExecutor.from_localcluster)
+    @_save
+    def dask_local(self, max_workers=_DEFAULT_CPUS, **kwargs):
+        return DaskExecutor.from_localcluster(max_workers=max_workers, **kwargs)
+
+    @wraps(DaskExecutor.from_cluster)
+    @_save
+    def dask_cluster(self, cluster):
+        return DaskExecutor.from_cluster(cluster)
+
+    del _save
+
+
+class RootCreator:
+    def __init__(self, project, config):
+        self._project = project
+        self._subcreators = {}
+        for name, (subcreator, subconfig) in config.items():
+            self._subcreators[name] = subcreator(project, subconfig)
+
+    def __dir__(self):
+        return tuple(self._subcreators.keys())
+
+    def register(self, name, subcreator, force=False):
+        """
+        Add a new creator.
+        """
+        if name not in self._subcreators or force:
+            self._subcreators[name] = subcreator
+        else:
+            raise ValueError(f"Already registered a creator under name {name}!")
+
+    def __getattr__(self, name):
+        try:
+            return self._subcreators[name]
+        except KeyError:
+            raise AttributeError(f"No creator of name {name} registered!")
+
+
+TASK_CONFIG = {
+    "AseStatic": "pyiron_contrib.tinybase.ase.AseStaticTask",
+    "AseMinimize": "pyiron_contrib.tinybase.ase.AseMinimizeTask",
+    "AseMD": "pyiron_contrib.tinybase.ase.AseMDTask",
+    "Murnaghan": "pyiron_contrib.tinybase.murn.MurnaghanTask",
```

Comment:
I always get nervous about maintainability when I see heavy use of hard-coded strings like this; is this necessary to avoid circular imports?

If not, is it possible to import the relevant objects and back the import path out directly from the objects? Then if they get moved or something in their path gets renamed, `TASK_CONFIG` will stay up-to-date by default. E.g. `from pyiron_contrib.tinybase.ase import AseStaticTask` and `"AseStatic": get_class_string(AseStaticTask)` where `def get_class_string(klass): return str(klass).split("'")[1]` (or whatever is most robust and readable in this function).